core/src/test/java/org/apache/calcite/test/RexProgramTest.java

```diff
@@ -2170,6 +2170,22 @@ private RexNode simplify(RexNode e) {
         is(false));
   }
 
+  @Test public void testIsAlwaysTrueAndFalse() {
+    final RelDataType booleanNullableType =
+        typeFactory.createTypeWithNullability(
+            typeFactory.createSqlType(SqlTypeName.BOOLEAN), true);
+    RexNode node = rexBuilder.makeInputRef(booleanNullableType, 0);
+
+    RexNode exp1 = rexBuilder.makeCall(SqlStdOperatorTable.IS_FALSE, isNotNull(isNull(node)));
+    assertEquals(exp1.isAlwaysTrue(), false);
```

Comment:
Please use `message` parameter for `assertEquals` or other means to make exception message clear.

Current one would be like "expected false, got true at line 2180.
It would be extremely hard to maintain if ever this test goes off the rails.

The same applies to all `assertEquals` in this test